plugins/PostProcessingPlugin/scripts/ColorMix.py

```diff
@@ -117,38 +126,44 @@ def execute(self, data):
         secondHeight = 0.0
         firstMix = 0.0
         SecondMix = 0.0
+        modelNumber = 0
 
         firstHeight = self.getSettingValueByKey("start_height")
         secondHeight = self.getSettingValueByKey("finish_height")
-        firstMix = self.getSettingValueByKey("mix_start_ratio")
-        SecondMix = self.getSettingValueByKey("mix_finish_ratio")
-
-        #locals
-        layer = 0
-
+        firstMix = self.getSettingValueByKey("mix_start")
+        SecondMix = self.getSettingValueByKey("mix_finish")
+        modelOfInterest = self.getSettingValueByKey("objectNumber")
+        
         #get layer height
-        layerHeight = .2
+        layerHeight = 0
         for active_layer in data:
             lines = active_layer.split("\n")
             for line in lines:
                 if ";Layer height: " in line:
                     layerHeight = self.getValue(line, ";Layer height: ", layerHeight)
                     break
+            if layerHeight != 0:
+                break
+
```

Comment:
Keep in mind that if the line "Layer height" does not exist in your GCode, the layerHeight will end up as 0 and the following lines will crash if unitsOfMeasurement is in mm, as you will divide by 0.

Reply:
Added default, good catch